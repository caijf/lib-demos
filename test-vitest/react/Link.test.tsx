import { expect, test } from 'vitest';
import { render, screen } from '@testing-library/react';
import { userEvent } from '@testing-library/user-event';
import Link from './Link.jsx';

test('Link changes the state when hovered', async () => {
  render(<Link page="http://antfu.me">Anthony Fu</Link>);

  const link = screen.getByText('Anthony Fu');

  expect(link).toHaveAccessibleName('Link is normal');

  await userEvent.hover(link);

  expect(link).toHaveAccessibleName('Link is hovered');

  await userEvent.unhover(link);

  expect(link).toHaveAccessibleName('Link is normal');
});

test('renders correctly', async () => {
  const result = render(<Link page="http://baidu.com">some text</Link>);
  const link = result.getByText('some text');

  expect(link).toMatchSnapshot();

  await userEvent.hover(link);
  expect(link).toMatchSnapshot();
});
