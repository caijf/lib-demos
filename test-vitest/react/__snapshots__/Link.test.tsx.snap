// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders correctly 1`] = `
<a
  aria-label="Link is normal"
  class="normal"
  href="http://baidu.com"
>
  some text
</a>
`;

exports[`renders correctly 2`] = `
<a
  aria-label="Link is hovered"
  class="hovered"
  href="http://baidu.com"
>
  some text
</a>
`;
